File: removeDuplicatesFromSortedArray.cpp
// given an integer array sorted
// remove duplicates in place such that each element appear only once
// the relative order of the elements should be kept the same
// return the new length (number of unique elements)

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> removeDuplicates(vector<int> arr) {
        vector <int> resp = {0};
        if (arr.size() == 0) return resp;
        
        int cnt = 0;
        vector <int> aux;
        for(int i = 1; i < arr.size(); i++) {
            if (arr[i] == arr[i-1]) cnt++;
            else aux.push_back(arr[i-1]);
        } 
        // retorna o array final sem repetições
        return aux;
        //return aux.size();
    }

    int main() {
        vector <string> arr;
        vector <int> arr2;
        string enter;
        // recebo o vector
        while (cin >> enter and (enter != "[") or (enter != "]") or (enter != ",")) {
            arr.push_back(enter);
        }    

        // transformar de string para inteiro
        for (int i = 0; i < arr.size(); i++) {
            int x = stoi(arr[i]);
            arr2[i] = x;
        }
        
        vector <int> final = removeDuplicates(arr2)
        
        // printar o vector final
        for (int i = 0; i < final.size(); i++) {
            cout << final[i] << " ";
        }
        return 0;
    }
};

File: Top150/removeDuplicatesFromSortedArray.py
# given array in increasing order, remove duplicates in place such that each element appear only once and return the new length
#  the relative order of elements should be kept the same
#  [1, 1, 2] => 2
#  [0,0,1,1,1,2,2,3,3,4] => 5
def removeDuplicates(arr):
    if len(arr) == 0:
        return 0
    # iterate the list 2 by 2, if the second one is eqaul to first one, remove the second
    for i in range(0, len(arr)-2):
        if i+1 < len(arr) and arr[i] == arr[i+1]:
            arr.remove(arr[i+1])
    
    # return the length of the list
    return len(arr)

# arr = [1,1,2]
# result = removeDuplicates(arr)
# print(result)

arr = []
# recebe o input
while True:
    elementArray = input().split(",")
    if elementArray != "":
        arr.append(elementArray)
        # print(arr)
    else:
        break
print(removeDuplicates(arr))
